File: web/src/test/java/com/kappa/web/support/ControllerTestBase.java
package com.kappa.web.support;

import org.junit.Before;
import org.springframework.test.web.servlet.MockMvc;
import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
import org.springframework.test.web.servlet.setup.MockMvcBuilders;
import org.springframework.web.context.WebApplicationContext;

import javax.inject.Inject;

public abstract class ControllerTestBase extends SpringTestBase {
    @Inject private WebApplicationContext webApplicationContext;
    private MockMvc mvc;

    @Before
    public void controllerTestBaseSetup() throws Exception {
        mvc = MockMvcBuilders.webAppContextSetup(webApplicationContext).build();
    }

    protected String get(String url) throws Exception {
        return mvc.perform(MockMvcRequestBuilders.get(url)).andReturn().getResponse().getContentAsString();
    }

    protected String post(String url) throws Exception {
        return mvc.perform(MockMvcRequestBuilders.post(url)).andReturn().getResponse().getContentAsString();
    }
}

File: web/src/test/kotlin/com/kappa/web/support/ControllerTestBase.kt
package com.kappa.web.support

import org.springframework.test.web.servlet.request.MockHttpServletRequestBuilder
import org.springframework.test.web.servlet.request.MockMvcRequestBuilders
import org.springframework.test.web.servlet.setup.MockMvcBuilders

abstract class ControllerTestBase : SpringTestBase() {
    protected abstract fun controller(): Any

    protected fun get(url: String): String = perform(MockMvcRequestBuilders.get(url))

    protected fun post(url: String): String = perform(MockMvcRequestBuilders.post(url))

    private fun perform(get: MockHttpServletRequestBuilder?) = mvc().perform(get).andReturn().response.contentAsString

    private fun mvc() = MockMvcBuilders.standaloneSetup(controller()).build()
}